cstubs_applicative/dune

```diff
@@ -0,0 +1,5 @@
+; (library
```

Comment:
Pulling this in will upset dune when this repo is used as a submodule, since there will be 2 versions of the `cstubs_applicative` library. I don't love duplicating the code here, but I think it's okay as long as we drop this dune file and make this copy bazel-only.

Can you delete this file?

Reply:
Done.